ludwig/datasets/goemotions/__init__.py

```diff
@@ -37,13 +38,20 @@ def __init__(self, cache_dir=DEFAULT_CACHE_LOCATION):
         super().__init__(dataset_name="goemotions", cache_dir=cache_dir)
 
     def read_file(self, filetype, filename):
-        file_df = pd.read_table(os.path.join(self.raw_dataset_path, filename), header=None)
+        file_df = pd.read_table(os.path.join(self.raw_dataset_path, filename),
+                                header=None)
         return file_df
 
     def process_downloaded_dataset(self):
         super(GoEmotions, self).process_downloaded_dataset()
         # format emotion ids to be a set of emotion ids vs. string
-        processed_df = pd.read_csv(os.path.join(self.processed_dataset_path, self.csv_filename))
+        processed_df = pd.read_csv(os.path.join(self.processed_dataset_path,
+                                                self.csv_filename))
         processed_df.columns = ['text', 'emotion_ids', 'comment_id', 'split']
-        processed_df['emotion_ids'] = processed_df['emotion_ids'].apply(lambda e_id: " ".join(e_id.split(",")))
-        processed_df.to_csv(os.path.join(self.processed_dataset_path, self.csv_filename), index=False)
+        processed_df['emotion_ids'] = processed_df['emotion_ids'].apply(
```

Comment:
be careful with lambda functions, they're difficult to debug in production mostly because you cant sometimes see whats happening inside them, for example in the scenario where e_id doesnt contain a comma, is that something that could happen?

Reply:
Python handles these scenarios pretty gracefully.  If the input does not have a comma, it returns a list with one element containing the entire string, which is what we want.